lib/FormatFactory.php

```diff
@@ -1,150 +1,55 @@
 <?php
-/**
- * This file is part of RSS-Bridge, a PHP project capable of generating RSS and
- * Atom feeds for websites that don't have one.
- *
- * For the full license information, please view the UNLICENSE file distributed
- * with this source code.
- *
- * @package	Core
- * @license	http://unlicense.org/ UNLICENSE
- * @link	https://github.com/rss-bridge/rss-bridge
- */
 
-/**
- * Factory class responsible for creating format objects from a given working
- * directory.
- *
- * This class is capable of:
- * - Locating format classes in the specified working directory (see {@see Format::$workingDir})
- * - Creating new format instances based on the format's name (see {@see Format::create()})
- *
- * The following example illustrates the intended use for this class.
- *
- * ```PHP
- * require_once __DIR__ . '/rssbridge.php';
- *
- * // Step 1: Set the working directory
- * Format::setWorkingDir(__DIR__ . '/../formats/');
- *
- * // Step 2: Create a new instance of a format object (based on the name)
- * $format = Format::create('Atom');
- * ```
- */
-class FormatFactory extends FactoryAbstract {
-	/**
-	 * Creates a new format object from the working directory.
-	 *
-	 * @throws \InvalidArgumentException if the requested format name is invalid.
-	 * @throws \Exception if the requested format file doesn't exist in the
-	 * working directory.
-	 * @param string $name Name of the format object.
-	 * @return object|bool The format object or false if the class is not instantiable.
-	 */
-	public function create($name){
-		if(!$this->isFormatName($name)) {
-			throw new \InvalidArgumentException('Format name invalid!');
-		}
+class FormatFactory
+{
+	private $folder;
+	private $formatNames;
 
-		$name = $this->sanitizeFormatName($name);
+	public function __construct(string $folder = PATH_LIB_FORMATS)
+	{
+		$this->folder = $folder;
 
-		if (is_null($name)) {
-			throw new \InvalidArgumentException('Unknown format given!');
+		// create format names
+		foreach(scandir($this->folder) as $file) {
```

Comment:
Any particular reason to collects the format names on construction, rather than just when `getFormatNames()` is called.

I guess it does not make much of a difference since scan and memory will be quite cheap for Formats directory but I do not really see any benefit to doing it this way either.

Reply:
The primary reason is to avoid the ad-hoc caching of the names with the use of a static var in the method. I guess it's pretty fast even without the ad-hoc cache due to fs cache. In general I don't mind doing work in constructors.